pcsx2/gui/Discord/DiscordRPC.cpp

```diff
@@ -0,0 +1,101 @@
+/*  PCSX2 - PS2 Emulator for PCs
+*  Copyright (C) 2002-2018  PCSX2 Dev Team
+*
+*  PCSX2 is free software: you can redistribute it and/or modify it under the terms
+*  of the GNU Lesser General Public License as published by the Free Software Found-
+*  ation, either version 3 of the License, or (at your option) any later version.
+*
+*  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+*  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+*  PURPOSE.  See the GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License along with PCSX2.
+*  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+// A Discord Rich Presence (DPRC) implementation.
+
+// Discord's official documentation for future expansion:
+// https://discordapp.com/developers/docs/rich-presence/how-to
+
+#include "PrecompiledHeader.h"   // memset, sprintf, integer shorthands
+#include "Discord/discord_rpc.h" // Discord's provided header for interfacing with their lib
+#include "DiscordRPC.h"          // Our customized header
+
+char *drpcGameTitle;
+u64 drpcStartTime;
+
+// Set the display name for the active game for the next update.
+// Pass dprcMenuStr or nullptr to reset to "In the menus".
+// * Does not immediately update Discord RPC, use drpcUpdate to do so
+// ** Side note, wx is a complete dumpster fire. Apparently passing a wxString in to a function crashes
+// the entire library, so we're just going to do wxString to char* conversions in core code and accept
+// that it adds a little more bulk.
+void drpcSetGame(char* newTitle) {
```

Comment:
I prefer the { begins on a new line to keep the code consistent on void functions.

```
void drpcSetGame(char* newTitle)
{
}
```